### PR TITLE
Add tests for WebDriver Async Session Creation

### PR DESCRIPTION
**Pull request by christian-bromann**
This patch adds tests for Async Session Creation, proposed in: https://github.com/w3c/webdriver/pull/1561

### CODE REVIEW

File: webdriver/tests/new_session/async.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+from .conftest import product, flatten
+
+from tests.support.asserts import assert_success
+from tests.new_session.support.create import valid_data
+
+
+@pytest.mark.parametrize("key,value", flatten(product(*item) for item in valid_data))
+def test_valid(new_session, add_browser_capabilities, key, value):
+    new_async_session_response, get_session_response = new_async_session(
```
Comment (whimboo) [inline webdriver/tests/new_session/async.py:13]: In the spec you define that as a new command with a different end-point. As such we should move related tests to a new sub directory called `new_async_session`.

File: webdriver/tests/new_session/async.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        {"capabilities": {
+            "alwaysMatch": add_browser_capabilities({key: value})}})
+    assert_success(new_async_session_response)
+    assert_success(get_session_response)
+
+@pytest.mark.parametrize("key,value", flatten(product(*item) for item in valid_data))
+def test_get_session(get_session, add_browser_capabilities, key, value):
+    bodyValue = get_session(
```
Comment (whimboo) [inline webdriver/tests/new_session/async.py:21]: Same here. Please create a new sub directory with the name `get_session` and move the tests over.

File: webdriver/tests/new_session/conftest.py [hunk 3]
```diff
@@ -11,6 +10,25 @@ def flatten(l):
...
+    def new_session(self, body):
+        return self.transport.send("POST", "session", body)
+
+    def get_session(self, session_id):
+        return self.transport.send("GET", "session/{}".format(session_id))
+
+    def delete_session(self, session_id):
+        self.transport.send("DELETE", "session/{}".format(session_id))
+
+    def new_session(self, body):
```
Comment (whimboo) [inline webdriver/tests/new_session/conftest.py:28]: Most likely you want `new_async_session` here.

**Review by whimboo — CHANGES_REQUESTED** [review]
Thanks Christian. Just some early feedback...

**Comment by AutomatedTester** [comment]
Thanks for the tests @christian-bromann . Please follow up on @whimboo's comments

**Comment by christian-bromann** [comment]
Thanks for the feedback all, will close this one given that I am working on modifications to this functionality. Will open a new PR later on.

**Pull request closed without merging**